core/src/web-plugin.ts

```diff
@@ -129,6 +129,11 @@ export class WebPlugin implements Plugin {
   load(): void {
     this.loaded = true;
   }
+
+  callNative(methodName: string, parameters?: any): Promise<any> {
+    const pluginName = this.constructor.name;
```

Comment:
We will need to come up with a more reliable way to retrieve the "plugin class".

I looked into this a little bit and this may involve some refactoring and a change in the API.

Reply:
Yeah, I've been trying to see if we could use the proxies in `runtime.ts` to add in the pluginName from there but I'm not making much progress. We could make another constructor that takes in the name optionally? Or we could just use `callNative` to mask the `Capacitor.nativePromise` call